skimage/measure/_regionprops.py

```diff
@@ -358,6 +407,94 @@ def __eq__(self, other):
         return True
 
 
+def _props_to_dict(regions, properties=('label', 'bbox'), separator='-', always_include_label=True):
+    """Convert image regions properties into a dictionary
+
+    Parameters
+    ----------
+    regions : (N,) list
+        List of _RegionProperties objects as returned by :func:`regionprops`
+    properties : tuple, optional
+        properties which the output need to have
+    separator : str, optional
+        Each element of non-scalar properties and is not listed in object_columns will
+        be put into its own column, with the index of that element separated from
+        the name by this separator.
+    always_include_label : bool, optional
+        allows autmatic adding of label to the output
+
+
+    Output
+    ------
+    out_dict : dict
+        Dictionary mapping property names to an array of values of that property, one per region.
+        This dictionary can be used as input to a pandas DataFrame, with keys mapping to columns.
+
+    Examples
+    --------
+    >>> from skimage import data, util, measure
+    >>> image = data.coins() > 110
+    >>> label_image = measure.label(image, connectivity=image.ndim)
+    >>> propslist = regionprops(label_image, image)
+    >>> props = to_dict(propslist, properties=['label', 'inertia_tensor', 'inertia_tensor_eigvals'])
+    >>> props
+    <output here>
+
+    The resulting dictionary can be directly passed to pandas, if installed, to
+    obtain a clean DataFrame::
+
+    >>> import pandas as pd  # doctest: +SKIP
+    >>> data = pd.DataFrame(props)  # doctest: +SKIP
+    >>> data.head()  # doctest: +SKIP
+    <output of data.head()>
+
+
+"""
+
+    object_columns = {'image', 'coords', 'convex_image', 'filled_image',
+               'intensity_image'}
+
+    if always_include_label and 'label' not in properties:
+        properties = ('label',) + properties
+
+    out = {}
+    n = len(regions)
+    for prop in properties:
+        dtype = COL_DTYPES[prop]
+        column_buffer = np.zeros(n, dtype=dtype)
+        try:
+            r = regions[0][prop]
+        except:
+            print(regions)
+        if np.isscalar(r) or prop in object_columns:
```

Comment:
Why is the following logic so complicated?  Is there a way we can unify the various cases?  If not, please document what we are dealing with here.

Reply:
@leGIT-bot I see now that without our offline discussions, this if statement is confusing. You should add a comment that this case happens is for properties that go in single columns, which is either those that are scalar properties, or object properties (those that cannot be raveled because they have different sizes for different objects). Probably you should add a Notes section in the docstring as well, explaining which columns are object columns.